aya-ebpf-macros/src/lsm.rs

```diff
@@ -17,38 +18,68 @@ impl Lsm {
         let item = syn::parse2(item)?;
         let mut args = syn::parse2(attrs)?;
         let hook = pop_string_arg(&mut args, "hook");
+        let cgroup = pop_bool_arg(&mut args, "cgroup");
         let sleepable = pop_bool_arg(&mut args, "sleepable");
+        
         err_on_unknown_args(&args)?;
         Ok(Lsm {
             item,
             hook,
+            cgroup,
             sleepable,
         })
     }
 
     pub(crate) fn expand(&self) -> Result<TokenStream> {
-        let section_prefix = if self.sleepable { "lsm.s" } else { "lsm" };
-        let section_name: Cow<'_, _> = if let Some(hook) = &self.hook {
-            format!("{}/{}", section_prefix, hook).into()
-        } else {
-            section_prefix.into()
-        };
-        let fn_vis = &self.item.vis;
-        let fn_name = self.item.sig.ident.clone();
-        let item = &self.item;
-        // LSM probes need to return an integer corresponding to the correct
-        // policy decision. Therefore we do not simply default to a return value
-        // of 0 as in other program types.
-        Ok(quote! {
-            #[no_mangle]
-            #[link_section = #section_name]
-            #fn_vis fn #fn_name(ctx: *mut ::core::ffi::c_void) -> i32 {
-                return #fn_name(::aya_ebpf::programs::LsmContext::new(ctx));
+        
+        if self.cgroup{
+            let section_name = if let Some(name) = &self.hook{
+                format!("lsm_cgroup/{}", name)
+            }else{
+                ("lsm_cgroup").to_owned()
+            };
+                
+            let fn_name = &self.item.sig.ident;
+            let item = &self.item;
 
-                #item
-            }
-        })
+            Ok(quote! {
+                #[no_mangle]
+                #[link_section = #section_name]
+                fn #fn_name(ctx: *mut ::core::ffi::c_void) -> i32 {
+                    return #fn_name(::aya_ebpf::programs::LsmContext::new(ctx));
+    
+                    #item
+                }
+            })
+
+        }else{
+            let section_prefix = if self.sleepable { "lsm.s" } else { "lsm" };
+            let section_name: Cow<'_, _> = if let Some(hook) = &self.hook {
+                format!("{}/{}", section_prefix, hook).into()
+            } else {
+                section_prefix.into()
+            };
+
+            let fn_vis = &self.item.vis;
+            let fn_name = self.item.sig.ident.clone();
+            let item = &self.item;
+
+            Ok(quote! {
+                #[no_mangle]
+                #[link_section = #section_name]
+                #fn_vis fn #fn_name(ctx: *mut ::core::ffi::c_void) -> i32 {
+                    return #fn_name(::aya_ebpf::programs::LsmContext::new(ctx));
+    
+                    #item
+                }
+            })
+        }
+        
     }
+
+     // LSM probes need to return an integer corresponding to the correct
+    // policy decision. Therefore we do not simply default to a return value
```

Comment:
There is something going wrong with your code formatting. Not only here. Can you fix it with rustfmt locally?